src/modules/schema.js

```diff
@@ -124,22 +100,13 @@ export default class Schema {
    * @param {object} json - The JSON schema to validate
    * @returns {Boolean}
    */
-  static validateSchema(json) {
-    let jsonSchemaSpec;
-    if (json['$schema']) {
-      if (json['$schema'].startsWith('http://json-schema.org/draft-07/schema')) {
-        jsonSchemaSpec = JSONSchema07;
-      } else {
-        // TODO: Fetch json['$schema'] (using http client) and convert function to async
-      }
-    } else {
-      return false;
-    }
-    // Fixme: Is the intention to throw error or return false on error
-    const result = validate(json, jsonSchemaSpec, {
-      throwError: true
+  static async validateSchema(json) {
+    // Get the JSON schema spec to check against.
+    const jsonSchemaSpec = await this.getJSONSchemaSpec(json);
+
+    validate(json, jsonSchemaSpec, {
```

Comment:
`return validate(json, ...` here - just incase user wants to see result and think its better than returning undefined